_pages/dev/lazy-loading-guide.md

```diff
@@ -0,0 +1,204 @@
+---
+title: Lazy Loading Guide
+---
+
+Lazy loading, also known as code splitting, lets you divide your JavaScript code into multiple chunks. The result is that you do not have to load all of the chunks when a user first accesses your storefront. Instead, chunks are only loaded when they are needed.
+
+Such an approach can substantially improve "Time To Interactive", especially in the case of complex web applications being accessed by low-end mobile devices.
+
+## Spartacus Approach to Lazy Loading
+
+Spartacus is mostly CMS driven, which means Spartacus does not use much of the built-in, route-based lazy loading that is offered out-of-the-box by Angular. Code splitting has to be done at app build time, although Spartacus cannot know what components or features a specific route can contain before actually loading them.
+
+As a result, Spartacus provides CMS-driven lazy loading in the following ways:
+
+- Lazy loading of CMS components
+- CMS-driven lazy loading of feature modules
+
+## General Concepts
+
+The following sections offer some important information about how lazing loading works in Spartacus.
+
+### Define Dynamic Imports Only in the Main Application
+
+Dynamic imports, a technique that is used to facilitate lazy loading and that also allows code splitting, can only be used in the main application. It is not possible to define dynamic imports in the prebuilt libraries.
+
+This is an unfortunate limitation, which results in some application code that must be added by customers. Although the amount of custom code is limited to the bare minimum, we'll add a feature in a future version of the schematics library to automatically add lazy loaded modules.
+
+### Avoiding Static Imports for Lazy Loaded Code
+
+To make code spitting possible, your static JavaScript code (the main app bundle) should not have any static imports to code that you want to lazy load. The builder will notice that the code is already included, and as a result, will not generate a separate chunk for it. This is especially important in the case of importing symbols from libraries.
+
+At the time of writing (Angular 9 and Angular 10), mixing static imports with dynamic imports for the same library entry point, even for distinct symbols, will break lazy loading and tree shaking for this library entry point. This includes the whole entry point statically into the build. For this reason, it is highly recommended you create separate entry points for the code that has to be loaded statically, and for the code that can be loaded lazily.
+
+### Configuration in Lazy Loaded Modules
+
+The lazy loaded configuration is not visible to other modules and does not contribute to the global configuration of the running application. This will be addressed in Spartacus 3.x with the concept of unified configuration. For Spartacus 2.1, you should import modules with configuration statically.
```

Comment:
> The lazy loaded configuration is not visible to other modules and does not contribute to the global configuration of the running application. 

If additional configuration is provided inside the lazy loaded module, this configuration will not be merged in the global application configuration. As a result, the configuration will not affect any component other than the lazy loaded components.

This limitation will be addressed in Spartacus 3.x with the concept of unified configuration. For Spartacus 2.x, you should import modules with configuration statically.

Reply:
👍